src/rx-query.ts

```diff
@@ -213,6 +204,7 @@ export class RxQueryBase<
         if (typeof newResultData === 'number') {
             this._result = {
                 docsData: [],
+                docsKeys: [],
```

Comment:
what is the purpose of this new field? I guess `Object.keys(docsMap)` isn't ordered? why not just `docsData.map((d)=>d.primary)`?

Reply:
Not strictly necessary, just a minor optimization. We only need the keys, and there is no reason to iterate over the result set later.

Reply:
but Object.keys() is super fast, no?

Reply:
It is still `O(n)` and not necessary.

Reply:
i will tie break here and say lets just keep docKeys